packages/lib/file-api-driver-amazon-s3.js

```diff
@@ -27,8 +28,18 @@ class FileApiDriverAmazonS3 {
 	}
 
 	hasErrorCode_(error, errorCode) {
-		if (!error || typeof error.code !== 'string') return false;
-		return error.code.indexOf(errorCode) >= 0;
+		if (!error || typeof error.name !== 'string') return false;
+		return error.name.indexOf(errorCode) >= 0;
+	}
+
+
+	streamToString_(stream){
+		return new Promise((resolve, reject) => {
+			const chunks = [];
+			stream.on("data", (chunk) => chunks.push(chunk));
+			stream.on("error", reject);
+			stream.on("end", () => resolve(Buffer.concat(chunks).toString("utf8")));
```

Comment:
If there's an error, are you sure that the "end" handler is not also being called? I had weird issues like this with streams that meant that both reject and resolve handlers ended up being called.

Reply:
lemme try and create that case. This is the "recommended" approach from the aws-sdk-v3 folks so gonna wrap my head around it more.

Reply:
This is resolved as the latest approach is using Response to handle this.